### DISCUSSION TITLE
psiDeepSpeech

### DISCUSSION THREAD

**Discussion by profwillie** [microsoft/psi#176] [Show and tell]
I have updated my \psi component for DeepSpeech.  It currently uses DeepSpeech 0.9.3 and now supports configurable model file name, scorer file name, beam width, and hot words.  The source code can be found [here](https://github.com/FandM-CARES/psiDeepSpeech), and the nuget should be available on nuget.org soon.

I do not have usage or install instructions yet, but it should be straightforward.  The most important (undocumented) step is to download a model file (and optionally a scorer file) at https://github.com/mozilla/DeepSpeech/releases.  After that you should be able to do the following:
```                
                var dsr = new DeepSpeechRecognizer(pipeline);

                audioSource.PipeTo(dsr.AudioIn);

                dsr.TextOut.Do(x => Console.WriteLine(x));
```

This currently has a Windows dependency because it uses the system voice activity detector.  If anyone wants to contribute a decent voice activity detector for Linux, I'd be happy to merge it in.

**Comment by sandrist** [comment]
This is fantastic, thanks for sharing!

+1 on the need for a good voice activity detector component for Linux. We have been on the lookout, but haven't found anything yet.
